_docs/ops/runbook/troubleshooting-logsearch.md

```diff
@@ -123,35 +130,101 @@ mutate {
+
+```sh
+export TIMECOP_REJECT_LESS_THAN_HOURS=$((180 * 24))
+```
+
+### Adjust the IAM role policy to allow the reindexing to read from the S3 bucket
+
+You must go in to the AWS console and adjust the IAM role policy for the environment ingestor you are restoring in, then
+attach the role policy to the ingestor VM you are working in.  When you find the IAM role policy, make sure the Allow Actions match this block:
+
+```
+"s3:PutObject",
+"s3:DeleteObject",
+"s3:GetObject",
+"s3:ListObjects",
+"s3:ListObjectsV2",
+"s3:ListObjectVersions",
+"s3:ListBucket"
+```
+
+Then make sure that Allow Resources matches this block:
+
+```
+"arn:${aws_partition}:s3:::logsearch-*/*",
+"arn:${aws_partition}:s3:::logsearch-*"
+```
+
+Save the role policy adjustments, then attach the role policy to the ingestor VM you are working in.
+
 ### Start the reindexing
 Run logstash passing in your edited configuration file:
 
 ```sh
 export JAVA_HOME=/var/vcap/packages/openjdk-11/jre
-rm /tmp/s3_import.sincedb
+rm /var/vcap/data/ingestor_syslog/tmp/s3_import.sincedb
 /var/vcap/packages/logstash/bin/logstash --path.config /var/vcap/data/ingestor_syslog/tmp/logstash-restore.config
 ```
 
-### Monitor progress
+### Monitor reindexing progress
 
-Logstash will run forever once started. Monitor the progress of the reindex, and stop logstash once the data has been reindexed. Progress can be monitored by tailing the sincedb file which logstash will update after each file it processes.
+Logstash will run forever once started. Monitor the progress of the reindex, and stop logstash once the data has been reindexed. Progress can be monitored by first setting turning the logstash reindexing to a background process and then by tailing the sincedb file which logstash will update after each file it processes:
 
 ```sh
+# While logstash process is running in your foreground:
+<Ctrl>+Z
+bg
+disown
+
+# Once the process is in the background and you're back at the terminal:
 watch cat /var/vcap/data/ingestor_syslog/tmp/s3_import.sincedb
 ```
 
-### Recreate ingestor
+When the proocess has finishing ingesting up to the point at which you are done, kill the background process:
+
+```sh
+# Find the pid (process ID) of the logstash process you put in the background:
+ps aux | grep <process name>
+
+# Terminate the process:
+kill <pid>
+```
+
+### Recreate ingestor and restore IAM role policy
 
 To restore the ingestor to known good configuration after the restore, recreate the VM:
 
 ```sh
 bosh -d logsearch recreate ingestor/0
 ```
 
+This will also remove the IAM role policy you added to the VM.  Finally, restore the IAM role policy itself.  Adjust the Allow Actions block to its former state:
+
+```
+"s3:PutObject",
+"s3:DeleteObject"
+```
+
+And adjust the Allow Resources block to its former state:
+
+```
+"arn:${aws_partition}:s3:::logsearch-*/*"
+```
+
+Now save the policy, and you're done!
+
```

Comment:
I think we should just reference terraform here, in case these permissions change in the future

Reply:
Done!